src/main/java/com/pinterest/secor/parser/MessageParser.java

```diff
@@ -60,7 +60,7 @@ public ParsedMessage parse(Message message) throws Exception {
         String[] partitions = extractPartitions(message);
```

Comment:
Added code for timestamp extraction based on configuration value and using it in partition generation.
Also added tests for the same in different parsers.